```
/var/lib/jenkins/jobs/dronin/workspace@2/Nodes/master/ground/gcs/src/plugins/debuggadget/debuggadgetfactory.cpp: In function ‘void customMessageHandler(QtMsgType, const QMessageLogContext&, const QString&)’:
/var/lib/jenkins/jobs/dronin/workspace@2/Nodes/master/ground/gcs/src/plugins/debuggadget/debuggadgetfactory.cpp:58:116: error: ‘level’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
     DebugEngine::getInstance()->message(level, msg, QString(context.file), context.line, QString(context.function));
```

--- a/ground/gcs/src/plugins/debuggadget/debuggadget.pri
+++ b/ground/gcs/src/plugins/debuggadget/debuggadget.pri
@@ -1,4 +1,1 @@
-include(../../gcsplugin.pri)
-include(../../plugins/coreplugin/coreplugin.pri)
-
 LIBS *= -l$$qtLibraryName(DebugGadget)

--- a/ground/gcs/src/plugins/debuggadget/debuggadgetfactory.cpp
+++ b/ground/gcs/src/plugins/debuggadget/debuggadgetfactory.cpp
@@ -34,7 +34,7 @@
 
 void customMessageHandler(QtMsgType type, const QMessageLogContext &context, const QString &msg)
 {
-    DebugEngine::Level level;
+    DebugEngine::Level level = DebugEngine::DEBUG;
     switch (type) {
     case QtDebugMsg:
         level = DebugEngine::DEBUG;

--- a/ground/gcs/src/plugins/debuggadget/debuggadgetwidget.cpp
+++ b/ground/gcs/src/plugins/debuggadget/debuggadgetwidget.cpp
@@ -113,6 +113,8 @@ void DebugGadgetWidget::message(DebugEngine::Level level, const QString &msg, co
     QString source;
 #ifdef QT_DEBUG // only display this extended info to devs
     source = QString("[%0:%1 %2]").arg(file).arg(line).arg(function);
+#else
+    Q_UNUSED(file); Q_UNUSED(line); Q_UNUSED(function);
 #endif
 
     m_config->plainTextEdit->setTextColor(color);

--- a/ground/gcs/src/plugins/uploader/uploader.pro
+++ b/ground/gcs/src/plugins/uploader/uploader.pro
@@ -4,6 +4,7 @@ DEFINES += UPLOADER_LIBRARY
 QT += svg widgets
 QT += testlib
 QT += network
+include(../../gcsplugin.pri)
 include(uploader_dependencies.pri)
 include(../../libs/glc_lib/glc_lib.pri)
 

--- a/ground/gcs/src/plugins/uploader/uploader_dependencies.pri
+++ b/ground/gcs/src/plugins/uploader/uploader_dependencies.pri
@@ -1,4 +1,3 @@
-include(../../gcsplugin.pri)
 include(../../plugins/coreplugin/coreplugin.pri)
 include(../../plugins/uavobjects/uavobjects.pri)
 include(../../plugins/uavtalk/uavtalk.pri)

--- a/ground/gcs/src/plugins/usagestatsgadget/usagestatsgadget.pro
+++ b/ground/gcs/src/plugins/usagestatsgadget/usagestatsgadget.pro
@@ -2,6 +2,7 @@ QT += network
 TEMPLATE = lib
 TARGET = UsageStats
 DEFINES += USAGESTATS_LIBRARY
+include(../../gcsplugin.pri)
 include(usagestats_dependencies.pri)
 HEADERS += \
     usagestatsplugin.h \

